TASVideos.WikiEngine/WikiModules.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace TASVideos.WikiEngine
+{
+	public static class WikiModules
+	{
+		public const string Awards = "awards";
+		public const string BrokenLinks = "brokenlinks";
+		public const string ActiveTab = "activetab";
+		public const string CurrentEditLink = "wikigetcurrenteditlink";
+		public const string DisplayMiniMovie = "displayminimovie";
+		public const string DisplayMovies = "displaymovie";
+		public const string EditorActivity = "editoractivity";
+		public const string FeedLog = "feedlog";
+		public const string FirstEditionTas = "firsteditiontas";
+		public const string Frames = "frames";
+		public const string FrontpageSubmissionList = "frontpagesubmissionlist";
+		public const string GameName = "gamename";
+		public const string GameSubPages = "gamesubpages";
+		public const string ListLanguages = "listlanguages";
+		public const string ListParents = "listparents";
+		public const string ListSubPages = "listsubpages";
+		public const string MediaPosts = "mediaposts";
+		public const string MoviesByAuthor = "moviesbyplayer";
+		public const string MoviesGameList = "moviesgamelist";
+		public const string MoviesList = "movieslist";
+		public const string PlatformAuthorList = "platformtaserlists";
+		public const string PlatformFramerates = "platformframerates";
+		public const string RejectedSubmissions = "rejectedsubmissions";
+		public const string SupportedMovieTypes = "submittableformats";
+		public const string TabularMovieList = "tabularmovielist";
+		public const string TimeSinceDate = "timesincedate";
+		public const string TopicFeed = "topicfeed";
+		public const string UnmirroredMovies = "unmirroredmovies";
+		public const string UserGetWikiName = "usergetwikiname";
+		public const string UserMovies = "usermovies";
+		public const string UserName = "user_name";
+		public const string Welcome = "welcome";
+		public const string WikiLink = "__wikilink";
+		public const string WikiOrphans = "wikiorphans";
+		public const string WikiTextChangeLog = "wikitextchangelog";
+		public const string WikiUsers = "wikiusers";
+		public const string Youtube = "youtube";
+
+		private static readonly IList<string> Modules = typeof(WikiModules)
+			.GetFields(BindingFlags.Public | BindingFlags.Static | BindingFlags.FlattenHierarchy)
+			.Where(fi => fi.IsLiteral && !fi.IsInitOnly)
+			.Select(fi => fi.GetRawConstantValue()?.ToString() ?? "")
+			.ToList();
+
+		public static bool TryGetModule(string name, out string moduleName)
+		{
+			var result = Modules.FirstOrDefault(s => s == name);
+			if (result == null)
+			{
+				moduleName = "";
+				return false;
+			}
+
+			moduleName = result;
```

Comment:
Is it intended that `moduleName` is exactly equal to `name`?  What's the point of that mapping?

Reply:
The point of the mapping is so that the public const string values can be used in the attribute, and you have that reference that shows which are in use.  

Since the ViewComponents are in an assembly we do not want referenced in the wiki assembly, something along these lines are necessary to tie them together with more than just magic strings.

The value must equal the markup in the code. The name of the const is arbituary.

 Though, it occurs to me that we need to make them case insensitive.  That is how it was before this refactor

Reply:
That all makes sense but it's still not explaining the two separate variables here.  What is `name` and what is `moduleName` and why are they separate variables if they always have the same value?

Reply:
moduleName is the text found in the wiki markup.  We match it up to a known wikiModule, and if there is a match, that module is executed

Reply:
But they're the same, the two values are exactly the same!  This method checks if the value is a valid one, but it doesn't change it at all.